RELEASE-NOTES.md

```diff
@@ -1,5 +1,41 @@
 # Prefect Release Notes
 
+## Release 2.13.5
+
```

Comment:
this release could have a bunch of highlights and they can go here

Reply:
I added notes about `pydantic>=2` but these should definitely not be the top highlights :D

Reply:
I added a header to your section @chrisguidry. Feel free to change it if needed

Reply:
I think that's perfect, TBH